src/cupbearer/detectors/statistical/mahalanobis_detector.py

```diff
@@ -6,30 +7,75 @@
 )
 
 
+def _pinv(C, rcond):
+    # Workaround for pinv not being supported on MPS
+    if C.device.type == "mps":
+        return torch.linalg.pinv(C.cpu(), rcond=rcond, hermitian=True).to(C.device)
+    return torch.linalg.pinv(C, rcond=rcond, hermitian=True)
+
+
+# From https://gist.github.com/chausies/011df759f167b17b5278264454fff379
+def norm_cdf(x):
+    return (1 + torch.erf(x / np.sqrt(2))) / 2
+
+
+def log_norm_cdf_helper(x):
+    a = 0.344
+    b = 5.334
```

Comment:
I get that the gist didn't have a source for these numbers, but they are still a bit suss for me. Also, precision is kinda low, though, probably doesn't matter as long as same numbers are used everytime.

Reply:
Never mind on the source, found it here from the comment on the gist https://en.wikipedia.org/wiki/Q-function#Bounds_and_approximations